src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -1727,15 +1727,38 @@ private static void AddBuildRequestToSubmission(BuildSubmission submission, int
 
         private static void AddProxyBuildRequestToSubmission(
             BuildSubmission submission,
-            int configurationId,
+            BuildRequestConfiguration configuration,
             ProxyTargets proxyTargets,
             int projectContextId)
         {
+            // Reverse the map so we can look up requested targets
+            // The ProxyTargetToRealTargetMap is "backwards" from how most users would want to use it and doesn't provide as much flexibility as it could if reversed.
+            // Unfortunately this is part of a public API so cannot easily change at this point.
+            Dictionary<string, string> realTargetsToProxyTargets = new(proxyTargets.ProxyTargetToRealTargetMap.Count, StringComparer.OrdinalIgnoreCase);
+            foreach (KeyValuePair<string, string> kvp in proxyTargets.ProxyTargetToRealTargetMap)
+            {
+                // In the case of multiple proxy targets pointing to the same real target, the last one wins. Another awkwardness of ProxyTargetToRealTargetMap being "backwards".
+                realTargetsToProxyTargets[kvp.Value] = kvp.Key;
+            }
```

Comment:
I'm thinking that this probably more logically fits into `ProxyTargets` class. The string comparison used would then be defined in the single spot (not prone to mismatches)